week1/645_set_mismatch.py

```diff
@@ -0,0 +1,11 @@
+# https://leetcode.com/problems/set-mismatch/
+def set_mismatch(nums: list[int]) -> list[int]:
+    freq_dict = {i + 1: 0 for i in range(len(nums))}
+    for n in nums:
+        freq_dict[n] += 1
+
+    freq_list = list(freq_dict.values())
```

Comment:
If you were to convert this to a list, maybe it would have been better if you create this already as a list from the beginning. You need an additional iteration to convert this into a list.

And one question, were you already aware that [dictionary iteration order from `values()`](https://stackoverflow.com/questions/2053021/is-the-order-of-a-python-dictionary-guaranteed-over-iterations) were guaranteed above Python 3.7? I was surprised that this code worked because dictionaries are implemented as hashmap and in general does not preserve the insertion order. I had to look up for the correct semantics.

Reply:
Oh yes, it's good to remind that the order of a python dictionary (like HashMap in Java) is not always guaranteed. I've been using the method blindly since I always used version > 3.7. Thank you!